Coding/fw/src/SerialAdapter.h

```diff
@@ -0,0 +1,122 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Adapters
+@{
+@file       SerialAdapter.h
+
+ESP32 Serial Adapter for Lawicel Protocol
+
+* @}
+***************************************************************************************************/
+#ifndef SERIAL_ADAPTER_H
+#define SERIAL_ADAPTER_H
+
+/* INCLUDES ***************************************************************************************/
+#include <Serialinterface.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* FORWARD DECLARATIONS ***************************************************************************/
+
+class SerialAdapter : public SerialInterface
+{
+public:
+  /* CONSTANTS ******************************************************************************/
+
+  /* TYPES **********************************************************************************/
+
+  /**
+  * Default Serial Adapter constructor.
+  */
+  SerialAdapter() : SerialInterface(), m_baudrate(115200)
+  {
+  }
+
+  /**
+  * Default Serial Adapter destructor.
+  */
+  ~SerialAdapter()
+  {
+  }
+
+  /**
+  * Initialize Module.
+  */
+  uint8_t begin()
+  {
+    Serial.begin(m_baudrate);
+    return 0;
+  }
+
+  /**
+  * Terminate Module
+  */
+  uint8_t end()
+  {
+    Serial.end();
+    return 0;
+  }
+
+  /**
+  * Sets the Baudrate for Serial Communication.
+  */
+  void setBaudrate(long _baudrate)
+  {
+    m_baudrate = _baudrate;
+    Serial.updateBaudRate(m_baudrate);
+  }
+
+  /**
+  * Reads the Serial Adapter into a buffer.
+  */
+  String read()
+  {
+    if (Serial.available() > 0){
+      return Serial.readStringUntil('\r');
```

Comment:
blocking call ? 
This is usually not a good thing. Need to discuss your runtime model.